backend/sirius-web-spring-collaborative/src/main/java/org/eclipse/sirius/web/spring/collaborative/representations/SubscriptionManager.java

```diff
@@ -42,20 +43,22 @@
 
     private final Many<IPayload> sink = Sinks.many().multicast().directBestEffort();
 
-    private final Map<String, Integer> username2subscriptionCount = new HashMap<>();
+    private final Map<String, AtomicInteger> username2subscriptionCount = new ConcurrentHashMap<>();
 
     @Override
     public void add(IInput input, String username) {
-        int subscriptionCount = this.username2subscriptionCount.getOrDefault(username, 0).intValue();
-        this.username2subscriptionCount.put(username, subscriptionCount + 1);
+        var subscriptionCount = this.username2subscriptionCount.getOrDefault(username, new AtomicInteger());
+        subscriptionCount.getAndIncrement();
+
+        this.username2subscriptionCount.put(username, subscriptionCount);
```

Comment:
Actually re-reading this I think this is not thread-safe either. From the initial state, two threads can enter `add`, both calling `getOrDefault`will return 2 `AtomicInteger`, increment them both to 1, and both but `1` into the map. The last one to call `put` will "win" but we still end with `1` for this username.

I think we should use `ConcurrentMap.computeIfAbsent()` instead of `getOrDefault` and then `put` to atomically get/initialize the `AtomicInteger` and then atomically mutate it.

Same logic for `remove`.